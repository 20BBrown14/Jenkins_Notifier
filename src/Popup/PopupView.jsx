import React from 'react';
import PropTypes from 'prop-types';
import Button from 'terra-button';
import JobInputComponent from './JobInputComponent';
import './PopupView.scss';

const propTypes = {
  /** Error message for input */
  errorMessage: PropTypes.string.isRequired,
  /** If input is invalid */
  formInvalid: PropTypes.bool,
  /** Function to validate input */
  fieldChanged: PropTypes.func.isRequired,
  /** Help message to display under input field */
  helpMessage: PropTypes.string,
  /** Value of the job input url input field */
  jobInputURL: PropTypes.string.isRequired,
  /** Handler for when validate is clicked */
  validateClickHandler: PropTypes.func,
  /** Handler for when confirm is clicked */
  confirmClickHandler: PropTypes.func,
};

const defaultProps = {
  formInvalid: false,
};

const PopupView = (props) => {
  const {
    errorMessage,
    formInvalid,
    fieldChanged,
    helpMessage,
    jobInputURL,
    validateClickHandler,
    confirmClickHandler,
  } = props;

  return (
    <div className="popupview-Window">
      <JobInputComponent
        errorMessage={errorMessage}
        formInvalid={formInvalid}
        fieldChanged={fieldChanged}
        helpMessage={helpMessage}
        jobInputURL={jobInputURL}
      />
      <Button
        style={{ margin: '0px 5px 0px 25px' }}
        text="Validate"
        onClick={validateClickHandler}
      />
      <Button
        style={{ margin: '0px 0px 0px 5px' }}
        text="Confirm"
        onClick={confirmClickHandler}
      />
    </div>
  );
};

PopupView.propTypes = propTypes;
PopupView.defaultProps = defaultProps;

export default PopupView;
