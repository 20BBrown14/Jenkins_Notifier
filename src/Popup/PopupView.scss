.popupview-Window {
  width: 650px;
  height: 450px;
  background-color: #fff991
}