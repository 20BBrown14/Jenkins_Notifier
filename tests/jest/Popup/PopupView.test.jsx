import React from 'react';
import PopupView from '../../../src/Popup/PopupView';

describe('PopupView', () => {
  it('should render a default view', () => {
    const testView = (shallow(
      <PopupView
        errorMessage="someMessage"
        handleValidation={() => {}}
        jobInputURL=""
        fieldChanged={() => {}}
      />,
    ));
    expect(testView).toMatchSnapshot();
  });
});
