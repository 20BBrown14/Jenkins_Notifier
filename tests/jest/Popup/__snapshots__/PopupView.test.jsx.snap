// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupView should render a default view 1`] = `
<div
  className="popupview-Window"
>
  <JobInputComponent
    errorMessage="someMessage"
    fieldChanged={[Function]}
    formInvalid={false}
    jobInputURL=""
  />
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    style={
      Object {
        "margin": "0px 5px 0px 25px",
      }
    }
    text="Validate"
    type="button"
    variant="neutral"
  />
  <Button
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isIconOnly={false}
    isReversed={false}
    style={
      Object {
        "margin": "0px 0px 0px 5px",
      }
    }
    text="Confirm"
    type="button"
    variant="neutral"
  />
</div>
`;
